iceoryx_hoofs/test/moduletests/test_unix_domain_sockets.cpp

```diff
@@ -69,45 +75,289 @@ class UnixDomainSocket_test : public Test
-    ASSERT_THAT(result.get_error(), Eq(IpcChannelError::INVALID_ARGUMENTS));
+    auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, "");
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkEmptySocketNameWithPathPrefixLeadsToInvalidChannelNameError)
+{
+    auto ret = UnixDomainSocket::unlinkIfExists("");
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkTooLongSocketNameWithPathPrefixLeadsToInvalidChannelNameError)
+{
+    UnixDomainSocket::UdsName_t longSocketName;
+    for (uint64_t i = 0U; i < UnixDomainSocket::LONGEST_VALID_NAME - strlen(UnixDomainSocket::PATH_PREFIX) + 1; ++i)
+    {
+        longSocketName.append(cxx::TruncateToCapacity, "o");
+    }
+    auto ret = UnixDomainSocket::unlinkIfExists(longSocketName);
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkExistingSocketIsSuccessful)
+{
+    UnixDomainSocket::UdsName_t socketFileName = UnixDomainSocket::PATH_PREFIX;
+    socketFileName.append(cxx::TruncateToCapacity, "iceoryx-hoofs-moduletest.socket");
+    ASSERT_TRUE(createTestSocket(socketFileName));
+    auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, socketFileName);
+    EXPECT_FALSE(ret.has_error());
 }
 
-TEST_F(UnixDomainSocket_test, UnlinkNonExistingWithInvalidNameLeadsToError)
+TEST_F(UnixDomainSocket_test, UnlinkExistingSocketWithPathPrefixLeadsIsSuccessful)
 {
-    auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, invalidName);
-    EXPECT_TRUE(ret.has_error());
-    ASSERT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+    UnixDomainSocket::UdsName_t socketFileName = "iceoryx-hoofs-moduletest.socket";
+    UnixDomainSocket::UdsName_t socketFileNameWithPrefix = UnixDomainSocket::PATH_PREFIX;
+    socketFileNameWithPrefix.append(cxx::TruncateToCapacity, socketFileName);
+    ASSERT_TRUE(createTestSocket(socketFileNameWithPrefix));
+    auto ret = UnixDomainSocket::unlinkIfExists(socketFileName);
+    EXPECT_FALSE(ret.has_error());
 }
 
-TEST_F(UnixDomainSocket_test, SendingOnServerLeadsToError)
+// the current contract of the unix domain socket is that a server can only receive
+// and the client can only send
+void sendOnServerLeadsToError(const sendCall_t& send)
 {
     cxx::string<10> message{"Foo"};
-    Duration duration{1_ms};
-    auto result = server.timedSend(message, duration);
+    auto result = send(message);
     EXPECT_TRUE(result.has_error());
-    ASSERT_THAT(result.get_error(), Eq(IpcChannelError::INTERNAL_LOGIC_ERROR));
+    EXPECT_THAT(result.get_error(), Eq(IpcChannelError::INTERNAL_LOGIC_ERROR));
+}
+
+TEST_F(UnixDomainSocket_test, TimedSendOnServerLeadsToError)
```

Comment:
Oh, how about naming this `ServerCanOnlyReceiveThereforeTimedSendLeadsToError` and similar with the other 3 tests?